### DISCUSSION TITLE
Get actual size beforehand

### DISCUSSION THREAD

**Discussion by NefixEstrada** [jam1garner/binrw#264] [Q&A]
Hello!

Is it possible to know the size that will have a struct beforehand?

So, having,

```rs
#[binrw]
struct A {
  first: u8,
  second: u16,
  third: u8,
}
```

Is it possible to know that the size that the struct will have will be `4` at comptime?

Could also be added as `#[brw(calc(SOMETHING))]` for another struct? Or the struct itself?

Thank you!

**Comment by csnover** [comment]
binrw does not currently calculate the size of a struct at compile time. You could annotate it yourself with e.g. `impl A { const SIZE: usize = 4; }`. What are you using this information for?

**Reply by NefixEstrada** [reply]
To calculate fields on structs.

I have a `MessageA` struct, and then a `Header` struct. The header has a field that is called `size`, which contains the size that `MessageA` is going to have.

Since a lot of times this information is already known at comptime, I think it would be useful if it could be added somehow in the library itself and not annotate each structure manually

Thanks!
